tests/cnf/core/network/metallb/tests/bgp-gracefulrestart.go

```diff
@@ -127,6 +144,157 @@ var _ = Describe("BGP Graceful Restart", Ordered, Label(tsparams.LabelGRTestCase
 		})
 
 	})
+
+	Context("with bfd", func() {
+		var (
+			bfdProfile   *metallb.BFDBuilder
+			bfdConfigMap *configmap.Builder
+		)
+
+		BeforeAll(func() {
+			By("Editing the MachineConfiguration CR named cluster to change node disruption policies")
+			updateMachineConfigurationNodeDisruptionPolicy()
+		})
+
+		BeforeEach(func() {
+			By("Creating BFD profile")
+			frrk8sPods, err := pod.List(APIClient, NetConfig.MlbOperatorNamespace, metav1.ListOptions{
+				LabelSelector: tsparams.FRRK8sDefaultLabel,
+			})
+			Expect(err).ToNot(HaveOccurred(), "Failed to list pods")
+			bfdProfile = createBFDProfileAndVerifyIfItsReady(frrk8sPods)
+
+			By("Creating MetalLb configMap")
+			bfdConfigMap = createConfigMap(tsparams.LocalBGPASN, ipv4NodeAddrList, false, true)
+		})
+		Context("single hop", func() {
+			BeforeEach(func() {
+				By("Creating BGP peer")
+				bgpPeer, err = metallb.NewBPGPeerBuilder(APIClient, "testpeer", NetConfig.MlbOperatorNamespace,
+					ipv4MetalLbIPList[0], tsparams.LocalBGPASN, tsparams.LocalBGPASN).
+					WithBFDProfile(bfdProfile.Definition.Name).WithPassword(tsparams.BGPPassword).
+					WithGracefulRestart(true).Create()
+				Expect(err).ToNot(HaveOccurred(), "Fail to create bgpPeer")
+
+				By("Creating FRR Pod with network and IP address")
+				frrPod = createFrrPod(
+					masterNodeList[0].Object.Name, bfdConfigMap.Object.Name, []string{}, staticIPAnnotation)
+
+				By("Checking that BGP and BFD sessions are established and up")
+				verifyMetalLbBFDAndBGPSessionsAreUPOnFrrPod(frrPod, ipv4NodeAddrList)
+
+				By("Checking that BGP GracefulRestart is enabled on the neighbors")
+				verifyGREnabledOnNeighbors(frrPod, ipv4NodeAddrList)
```

Comment:
Do we check only single stack and ipv4? no ipv6 or dualstack?

Reply:
Yes, this was addressed during the high-level test design phase:
`IPv6 testing should be added once the CI clusters could support dual-stack deployment`

We are still facing the IPv6 cluster limitation, which is expected to be resolved soon by our new QE CI team.

Reply:
Okay, it would be nice to consider how test will be transition for single ipv6 and dual-stack testing, at least for the new tests.